File: ExtremeCore/Classes and Modules/fileHash.vb
Imports System.IO
Imports System.Security.Cryptography

Public Module FileHash
    Public Function GetFileHash(fileName As String) As String
        Dim sha = SHA1.Create()

        If File.Exists(fileName) = False Then
            Return False
        End If

        Using fs As FileStream = File.Open(fileName, FileMode.Open)
            Return Convert.ToBase64String(sha.ComputeHash(fs))
        End Using
    End Function
End Module

File: ExtremeStudio.Core/Classes and Modules/fileHash.cs
using System;
using System.IO;
using System.Security.Cryptography;


namespace ExtremeStudio.Core.Modules
{
	public sealed class FileHashGenerator
	{
		public static string GetFileHash(string fileName)
		{
			var sha = SHA1.Create();
			
			if (File.Exists(fileName) == false)
			{
				return System.Convert.ToString( false);
			}
			
			//To skipe file being used shit.
			byte[] fle = File.ReadAllBytes(fileName);
			return Convert.ToBase64String(sha.ComputeHash(fle));
		}
	}
}
